### PR TITLE
Jedno slovo za Drakulu

### PR DESCRIPTION
**Pull request by ugljesak**
evo cak i bez data seta :)

### CODE REVIEW

File: e.csv
Comment (Drakula44) [inline e.csv:1]: Ovaj falj treba da se izbrise

File: model.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+from matplotlib.pylab import f
+import numpy as np
+import pandas as pd
+import matplotlib as plib
+import torch
+from torch import nn
+
+class Generator(nn.Module):
+    def __init__(self, color_dim, out_dim, device):
```
Comment (Drakula44) [inline model.py]: color_dim nije bas opis ove promenljive vise da je mozda nazoves da je to dimenzija filtera ili broj kanala suma/embedinga kasnije

File: model.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        self.device = device
+        self.kernel_size = 4
+        super().__init__()
+        ngf = 32
+        #self.kernel = nn.Parametar(torch.randn(broj_slova, z_dim))
+        self.gen = nn.Sequential(            
+            nn.ConvTranspose2d(color_dim * 100, ngf * 8, self.kernel_size, 1, 0, bias=False, device=self.device),
```
Comment (Drakula44) [inline model.py]: a izbaci ove device=self.device iz konstruktora nebi trebalo da je potrebno ako jeste vraticemo naknadno

File: model.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            nn.BatchNorm2d(ngf * 2),
+            nn.ReLU(True),
+            
+            nn.ConvTranspose2d( ngf * 2, ngf, self.kernel_size, 2, 1, bias=False, device=self.device),
+            nn.BatchNorm2d(ngf),
+            nn.ReLU(True),
+            
+            nn.ConvTranspose2d( ngf, out_dim, self.kernel_size, 2, 1, bias=False ,device=self.device),
+            nn.Tanh()
+        ).to(self.device)
```
Comment (Drakula44) [inline model.py]: i ovaj to.(device)

File: model.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    def forward(self, noise):
+        #z = noise/2* self.kernel[slovo]
+        return self.gen(noise)
+        
+class Discriminator(nn.Module):
+    def __init__(self, color_dim, device):
```
Comment (Drakula44) [inline model.py]: color_dim ovde nije koriscen u ostatku konstruktora i isto moze da bolje da se zove input_chaneels dim ili nesto tako

File: model.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            nn.LeakyReLU(0.2, inplace=True),
+            
+            nn.Conv2d(ndf * 4, ndf * 8, self.kernel_size, 2, 1, bias=False, device=self.device),
+            nn.BatchNorm2d(ndf * 8),
+            nn.LeakyReLU(0.2, inplace=True),
+            
+            nn.Conv2d(ndf * 8, color_dim, self.kernel_size, 1, 0, bias=False, device=self.device),
+            nn.Sigmoid()
+
+      ).to(self.device)
```
Comment (Drakula44) [inline model.py]: i isto ove device poizbacuj

File: train.py [hunk 6]
```diff
@@ -1,52 +1,67 @@
...
+writer = SummaryWriter()
+gan.load_state_dict(torch.load('model_epoch_9.pt'))
 
-gen = Generator(img_dim, z_dim).to(device)
-disc = Discriminator(z_dim).to(device)
-z = torch.randn(batch_size, z_dim).to(device)
-gen_opt = torch.optim.Adam(gen.parameters(), lr=learn_rate)
-disc_opt = torch.optim.Adam(disc.parameters(), lr=learn_rate)
-criterion = nn.BCELoss()
+for epoch in range(10, num_epochs):
```
Comment (Drakula44) [inline train.py:24]: sto krece od 10 epohe? moze da se u .pt fajlu sacuva epoha iz koje je krenuo i onda da se krene od te epohe

File: train.py [hunk 7]
```diff
@@ -1,52 +1,67 @@
...
+    batch_idx = 0
```
Comment (Drakula44) [inline train.py:25]: mozes da se stavi enumerate oko for loop-a vise je pythonic nacin kucanja

File: train.py [hunk 8]
```diff
@@ -1,52 +1,67 @@
...
+        gan.gen_opt.zero_grad()
+        output = gan.disc(fake.view(curr_batch_size, -1))
+        
+        lossG = gan.criterion(output, torch.ones_like(output))
         lossG.backward()
-        gen_opt.step()
+        
+        gan.gen_opt.step()
 
+        for i in range(fake.size(0)):
```
Comment (Drakula44) [inline train.py:55]: nit: fake.size(0) je current_batch_size vise za lepotu koda samo

File: train_e.py
Comment (Drakula44) [inline train_e.py:1]: Ok nisam video da sam komentarisao train.py a ne train_e.py tako da svi komentari za train vaze i za train_e

File: train_e.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    for image, label in train_dataloader:
+        trainer += 1
+
+        curr_batch_size = len(image)
+        ### Train Discriminator: max log(D(x)) + log(1 - D(G(z)))
+        gan.disc_opt.zero_grad()
+        gan.gen_opt.zero_grad()
+        
+        noise = torch.randn(curr_batch_size, input_channels * 100, z_dim, z_dim).to(device)
```
Comment (Drakula44) [inline train_e.py]: Izdvojiti 100 kao konstantu

File: train_e.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        noise = noise / 2 + 0.5
```
Comment (Drakula44) [inline train_e.py]: I ovo skaliranje isto izdvojiti parametre

File: model.py (hunk 2 above)
Comment (ugljesak) [reply model.py]: e jesmo u paperspace-u samo smo ovde zaboravili sredjeno sad

File: model.py (hunk 3 above)
Comment (ugljesak) [reply model.py]: to mi milomir reko za svaki sluc da staavimo jer nece graficku da inicijalizuje

File: model.py (hunk 4 above)
Comment (ugljesak) [reply model.py]: e jeste na kraju koriscen, kao output_channels samo cu ga preimenujem

File: model.py (hunk 1 above)
Comment (ugljesak) [reply model.py]: reseno

File: model.py (hunk 5 above)
Comment (ugljesak) [reply model.py]: 👌

File: train.py (hunk 6 above)
Comment (ugljesak) [reply train.py:24]: e pogresan fajl trebas ovaj sa eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

File: train.py (hunk 7 above)
Comment (ugljesak) [reply train.py:25]: vazi

File: train.py (hunk 8 above)
Comment (ugljesak) [reply train.py:55]: vazi

File: train_e.py (hunk 9 above)
Comment (ugljesak) [reply train_e.py]: moze

File: train_e.py (hunk 10 above)
Comment (ugljesak) [reply train_e.py]: no problem

File: train_e.py
Comment (ugljesak) [reply train_e.py:1]: jesjes

File: e.csv
Comment (ugljesak) [reply e.csv:1]: : (
